zebra-network/src/protocol/internal/request.rs

```diff
@@ -39,11 +42,30 @@ pub enum Request {
     /// didn't start with a `Vec` but with, e.g., an iterator, they can collect
     /// directly into a `HashSet` and save work.
     ///
+    /// If this requests a recently-advertised block, the peer set will make a
+    /// best-effort attempt to route the request to a peer that advertised the
+    /// block. In this case, it's best to use a request set of size 1.
```

Comment:
Isn't the size a requirement?
```suggestion
    /// block. This routing is only used for request sets of size 1.
```

Reply:
Yes, good catch, I wrote the comment before dropping the more complex logic.